Packs/CortexAttackSurfaceManagement/Scripts/InferWhetherServiceIsDev/InferWhetherServiceIsDev.py

```diff
@@ -0,0 +1,149 @@
+import traceback
+from collections.abc import Mapping
+from typing import Any, Dict, List
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def _canonicalize_string(in_str: str) -> str:
+    """
+    Converts `in_str` to a canonical-for-dev-evaluation version of itself through:
+    * lowercasing
+    * stripping whitespace (space and tab)
+    * stripping quotation marks (" and ')
+    """
+    return in_str.lower().strip(' \t"\'')
+
+
+def is_dev_according_to_key_value_pairs(observed_key_value_pairs: List[Dict[str, Any]]) -> bool:
+    """
+    Returns whether the key-value pairs in `observed_key_value_pairs` explicitly & solely contain
+    an indication of "dev"ness. Indicators of "dev"ness were learned by AI from a large
+    corpus of canonically tagged infrastructure, designed to capture many ways in which
+    "dev" infrastructure might manifest within different operating environments.
+
+    If `observed_key_value_pairs` has indicators of both prod AND dev, returns False;
+    `observed_key_value_pairs` does not describe a definitively "development" role server.
+
+    Args:
+        observed_key_value_pairs (List[Dict[str, Any]]): list of key-value dictionaries;
+            each dictionary within the list must contain the keys "Key" and "Value";
+            the values are arbitrary
+            Example value for `observed_key_value_pairs`:
+                [{"Key": "env", "Source": "AWS", "Value": "dev"},
+                 {"Key": "Name", "Source": "AWS", "Value": "ssh-ec2-machine-name"}]
+
+    Returns:
+        bool: whether `observed_key_value_pairs` indicates that the described system is
+            used solely for development (has development indicators and lacks any
+            production indicators)
+    """
+    EXACT_DEV_MATCH = ["dv", "noprod", "np", "ppe"]
```

Comment:
They usually want all Constants on the top of the file, I think we should move these

Reply:
can we wait for them to complain? they are local constants to the method implementation, not to the file, so I don't think moving them would be good engineering.

Reply:
okay, i am fine with that

Reply:
something else that strikes me -- if you think they'll object and we can't win by arguing appropriate variable scope, then we could change the casing on the variables so they aren't visible uppercase constants.